src/Codeception/Application.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Codeception;
+
+use Codeception\Exception\ConfigurationException;
+use Symfony\Component\Console\Application as BaseApplication;
+use Symfony\Component\Console\Output\ConsoleOutput;
+use Codeception\Lib\Interfaces\CustomCommand;
+
+class Application extends BaseApplication
+{
+    /**
+     * Register commands from a config file
+     *
+     *  extensions:
+     *      commands:
+     *          - project\MyNewCommand
+     *
+     */
+    public function registerCustomCommands()
+    {
+        $config = Configuration::config();
```

Comment:
Somehow you should get a value from `-c` option if custom config is specified. Otherwise that won't work in predictable way